packages/eslint-plugin-svelte/src/rules/prefer-const.ts

```diff
@@ -0,0 +1,81 @@
+import type { TSESTree } from '@typescript-eslint/types';
+
+import { createRule } from '../utils/index.js';
+import { defineWrapperListener, getCoreRule } from '../utils/eslint-core.js';
+
+const coreRule = getCoreRule('prefer-const');
+
+/**
+ * Finds and returns the callee of a declaration node within variable declarations or object patterns.
+ */
+function findDeclarationCallee(node: TSESTree.Expression) {
+	const { parent } = node;
+	if (parent.type === 'VariableDeclarator' && parent.init?.type === 'CallExpression') {
+		return parent.init.callee;
+	}
+
+	return null;
+}
+
+/**
+ * Determines if a declaration should be skipped in the const preference analysis.
+ * Specifically checks for Svelte's state management utilities ($state, $props, $derived).
```

Comment:
```suggestion
 * Specifically checks for Svelte's state management utilities ($props, $derived).
```

I think here doesn't handle `$state`.